web/src/copy_and_paste.js

```diff
@@ -573,6 +574,35 @@ export function maybe_transform_html(html, text) {
     return html;
 }
 
+/**
+ * Try to convert the `text` to the `#**stream>topic**` syntax and optionally
+ * add the original text to the undo stack of the textarea
+ * @param {string} text
+ * @param {boolean} undo_to_original_url whether to add the original text to the undo stack
+ * @returns the `text` transformed to `#**stream>topic**` syntax if the `text`
+ * is a valid stream topic url, else the untransformed text as received in the argument
+ */
+export function try_stream_topic_syntax_text(text, undo_to_original_url) {
+    const stream_topic = hash_util.decode_stream_topic_from_url(text);
+    if (!stream_topic) {
+        return text;
+    }
+    if (undo_to_original_url) {
+        const text_area = $("textarea#compose-textarea").get(0);
+        text_area.focus();
+        const init_cursor_pos = text_area.selectionStart;
+        document.execCommand("insertText", false, text);
+        const new_cursor_pos = text_area.selectionStart;
+        text_area.setSelectionRange(init_cursor_pos, new_cursor_pos);
```

Comment:
This use of `execCommand` is I think deprecated? https://developer.mozilla.org/en-US/docs/Web/API/Document/execCommand

In any case, our general policy is for any manipulation of the compose area use the `compose_ui` module and its `text-field-edit` library, and we should follow that here as well.

Reply:
Sure, I'll switch to that. 
There were a few places in the codebase where `execCommand` is still used.